apis/shared-dimensions/lib/shapeToQuery.ts

```diff
@@ -0,0 +1,93 @@
+import onetime from 'onetime'
+import { md } from '@cube-creator/core/namespace'
+import { GraphPointer } from 'clownface'
+import { isGraphPointer } from 'is-graph-pointer'
+import { hydra } from '@tpluscode/rdf-ns-builders'
+import { Parameters, PropertyShape } from '@hydrofoil/shape-to-query/model/constraint/ConstraintComponent'
+import namespace from '@rdfjs/namespace'
+import { sparql } from '@tpluscode/sparql-builder'
+import env from './env'
+
+/*
+ The @hydrofoil/shape-to-query is an ES Module and as long as we compile to CJS, it needs to be loaded dynamically
+ using this ugly TS hack
+ */
+// eslint-disable-next-line no-new-func
+const _importDynamic = new Function('modulePath', 'return import(modulePath)')
+
+export default async function shapeToQuery() {
+  await setup()
+
+  const { constructQuery, deleteQuery } = await _importDynamic('@hydrofoil/shape-to-query')
+
+  return {
+    constructQuery,
+    deleteQuery,
+  }
+}
+
+const setup = onetime(async () => {
+  await defineConstraintComponents()
+})
+
+async function defineConstraintComponents() {
+  const { ConstraintComponent } = await _importDynamic('@hydrofoil/shape-to-query/model/constraint/ConstraintComponent.js')
+  const { constraintComponents } = await _importDynamic('@hydrofoil/shape-to-query/model/constraint/index.js')
+  const { PatternConstraintComponent } = await _importDynamic('@hydrofoil/shape-to-query/model/constraint/pattern.js')
+
+  constraintComponents.set(md.FreeTextSearchConstraintComponent, class TextSearch extends ConstraintComponent {
```

Comment:
Here is an implementation of SHACL node expression which generates full-text search patterns

In Stardog/Fuseki, vendor-specific constructs are used. Otherwise it falls back to plain REGEX